format.go

```diff
@@ -31,6 +32,10 @@ func (f *Formatter) Format(o interface{}, fields []string) (interface{}, error)
 	if len(fields) == 0 {
 		return o, nil
 	}
+	err := detectDuplicateFields(fields)
```

Comment:
Checking for duplicates in every call to Format seems unnecessary : once the formatter is built/cached, no need to recheck for duplicates. You probably could check for it either just before building the formatter (around line 56 in your version of the file) or (maybe better) in structBuilder.build